### PR TITLE
Billing records pages

### PR DESCRIPTION
**Pull request by aaronk**
Setup IFXBillingRecordList to display billing records
CalculateBillingMonth.vue allows facility admins to calculate billing records and see ProductUsageProcessing results.  Can be used as-is in facility applications

### CODE REVIEW

File: src/components/billingRecord/IFXCalculateBillingMonth.vue [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+      }
+      return !this.usagesPreviouslyBilled() || this.recalculate
+    },
+    hasFinalBillingRecord(usage) {
+      return usage.billingRecords && usage.billingRecords.some((br) => br.currentState === 'FINAL')
+    },
+    usagesPreviouslyBilled() {
+      // Returns true if any of the usages has billingRecords
+      if (this.usages) {
+        return this.usages.some((usage) => usage.billingRecords && usage.billingRecords.length)
```
Comment (luthian) [inline src/components/billingRecord/IFXCalculateBillingMonth.vue]: You can also do this
```suggestion
        return this.usages.some((usage) => usage?.billingRecords.length)
```

File: src/components/billingRecord/IFXCalculateBillingMonth.vue [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+  methods: {
+    ...mapActions(['showMessage']),
+    canCalculate() {
+      if (this.usages && this.usages.some((usage) => this.hasFinalBillingRecord(usage))) {
+        return false
+      }
+      return !this.usagesPreviouslyBilled() || this.recalculate
+    },
+    hasFinalBillingRecord(usage) {
+      return usage.billingRecords && usage.billingRecords.some((br) => br.currentState === 'FINAL')
```
Comment (luthian) [inline src/components/billingRecord/IFXCalculateBillingMonth.vue]: This should produce the same result but might not be as clear. Your call.
```suggestion
      return usage?.billingRecords.some((br) => br.currentState === 'FINAL')
```

File: src/components/billingRecord/IFXCalculateBillingMonth.vue [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+      if (this.recalculate) {
+        return 'Recalculate billing records'
+      }
+      return 'Calculate billing records'
+    }
+  },
+  methods: {
+    ...mapActions(['showMessage']),
+    canCalculate() {
+      if (this.usages && this.usages.some((usage) => this.hasFinalBillingRecord(usage))) {
```
Comment (luthian) [inline src/components/billingRecord/IFXCalculateBillingMonth.vue]: This should produce the same result but might not be as clear. Your call.
```suggestion
      if (this.usages?.some((usage) => this.hasFinalBillingRecord(usage))) {
```

File: src/components/billingRecord/IFXCalculateBillingMonth.vue [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+      })
+  }
+}
+</script>
+
+<template>
+  <v-container v-if="!isLoading">
+    <IFXPageHeader>
+      <template #title>Calculate billing month</template>
+      <template #actions>
```
Comment (luthian) [inline src/components/billingRecord/IFXCalculateBillingMonth.vue]: Should this be here if it is empty?

Comment (aaronk) [reply src/components/billingRecord/IFXCalculateBillingMonth.vue]: Will remove

**Pull request merged**